events/iot.go

```diff
@@ -1,34 +1,57 @@
 package events
 
-// IoTCustomAuthorizerRequest contains data coming in to a custom IoT device gateway authorizer function.
-type IoTCustomAuthorizerRequest struct {
-	HTTPContext        *IoTHTTPContext `json:"httpContext,omitempty"`
-	MQTTContext        *IoTMQTTContext `json:"mqttContext,omitempty"`
-	TLSContext         *IoTTLSContext  `json:"tlsContext,omitempty"`
-	AuthorizationToken string          `json:"token"`
-	TokenSignature     string          `json:"tokenSignature"`
+// IoTCoreCustomAuthorizerRequest represents the request to an IoT Core custom authorizer.
+// See https://docs.aws.amazon.com/iot/latest/developerguide/config-custom-auth.html
+type IoTCoreCustomAuthorizerRequest struct {
+	Token              string                     `json:"token"`
+	SignatureVerified  bool                       `json:"signatureVerified"`
+	Protocols          []string                   `json:"protocols"`
+	ProtocolData       *IoTCoreProtocolData       `json:"protocolData,omitempty"`
+	ConnectionMetadata *IoTCoreConnectionMetadata `json:"connectionMetadata,omitempty"`
 }
 
-type IoTHTTPContext struct {
+type IoTCoreProtocolData struct {
+	TLS  *IoTCoreTLSContext  `json:"tls,omitempty"`
+	HTTP *IoTCoreHTTPContext `json:"http,omitempty"`
+	MQTT *IoTCoreMQTTContext `json:"mqtt,omitempty"`
+}
+
+type IoTCoreTLSContext struct {
+	ServerName string `json:"serverName"`
+}
+
+type IoTCoreHTTPContext struct {
 	Headers     map[string]string `json:"headers,omitempty"`
 	QueryString string            `json:"queryString"`
 }
 
-type IoTMQTTContext struct {
+type IoTCoreMQTTContext struct {
 	ClientID string `json:"clientId"`
 	Password []byte `json:"password"`
 	Username string `json:"username"`
 }
 
-type IoTTLSContext struct {
-	ServerName string `json:"serverName"`
+type IoTCoreConnectionMetadata struct {
+	ID string `json:"id"`
+}
+
+// IoTCoreCustomAuthorizerResponse represents the response from an IoT Core custom authorizer.
+// See https://docs.aws.amazon.com/iot/latest/developerguide/config-custom-auth.html
+type IoTCoreCustomAuthorizerResponse struct {
+	IsAuthenticated          bool                     `json:"isAuthenticated"`
+	PrincipalID              string                   `json:"principalId"`
+	DisconnectAfterInSeconds uint32                   `json:"disconnectAfterInSeconds"`
+	RefreshAfterInSeconds    uint32                   `json:"refreshAfterInSeconds"`
+	PolicyDocuments          []*IoTCorePolicyDocument `json:"policyDocuments"`
+}
+
+type IoTCorePolicyDocument struct {
```

Comment:
This model duplicates `APIGatewayCustomAuthorizerPolicy`.

Reply:
I feel it's pretty strange to have the IoT Core authorizer struct have a dependency on the API Gateway struct. Even if they are the same struct, the names are rather jarring.
What I will do is make a copy of `APIGatewayCustomAuthorizerPolicy` named `IAMPolicyDocument` and make the former an alias of the latter. That we we don't break backwards compatibility, but new events aren't forced to reference an API Gateway struct.